base/shared/src/main/scala/scalaz/tc/foldable.scala

```diff
@@ -6,13 +6,24 @@ import scala.List
 import scala.language.experimental.macros
 
 trait FoldableClass[F[_]] {
+  /**
+  Applies a binary operator to a start value and all elements of this sequence, going left to right.
+  Note: this will not terminate for infinite-sized collections. For Example, Stream.
+  */
   def foldLeft[A, B](fa: F[A], z: B)(f: (B, A) => B): B
-
+  /**
+  Applies a binary operator to all elements of this list and a start value, going right to left.
+  */
   def foldRight[A, B](fa: F[A], z: => B)(f: (A, => B) => B): B
```

Comment:
Adding some extra comment noting that `foldRight` and `foldMap`, in contrast to `foldLeft`, will evaluate lazily would be nice. Be sure to state what is and is not evaluated according to which semantic